### ISSUE TITLE
[BUG] Rules with konveyor.io/source or konveyor.io/target wildcard match labels should match on any target

### ISSUE THREAD

**Issue by pranavgaikwad** [konveyor/analyzer-lsp#344]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

latest

### Priority

Major

### Current Behavior

The rules with these wildcard labels are skipped

### Expected Behavior

The rules should be matched

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Create a rule with label `konveyor.io/source`
2. Pass `--label-selector=konveyor.io/source=eap` as input when running analysis.

### Environment

```markdown
- **Analyzer Image SHA** latest
- **JDTLS  base image SHA** latest
- **Provider Settings JSON** latest
```
